09-maze_solver/progress/internal/solver/explore.go

```diff
@@ -31,15 +31,8 @@ func (s *Solver) listenToBranches() {
 }
 
 // explore one path and publish to the s.pathsToExplore channel any branch we discover that we don't take.
-func (s *Solver) explore(pathToBranch []image.Point) {
-	// A path starts at the entrance and has stepped into the maze
-	// for at least 1 pixel, for a total of 2 pixels minimum.
-	if len(pathToBranch) < 2 {
-		return
-	}
-
-	pos := pathToBranch[len(pathToBranch)-1]
-	previous := pathToBranch[len(pathToBranch)-2]
+func (s *Solver) explore(pathToBranch *path) {
+	pos := pathToBranch.at
```

Comment:
Are we sure that the param is not nil? Do we need to explain in the book that it's not exposed and we trust the next devs inside the package to test their code?

Reply:
good point.